jni/src/jni_util.cpp

```diff
@@ -294,6 +294,40 @@ void knn_jni::JNIUtil::Convert2dJavaObjectArrayAndStoreToByteVector(JNIEnv *env,
     env->DeleteLocalRef(array2dJ);
 }
 
+void knn_jni::JNIUtil::Convert2dJavaObjectArrayAndStoreToByteVector(JNIEnv *env, jobjectArray array2dJ,
+                                                                     int dim, std::vector<int8_t> *vect) {
+
+    if (array2dJ == nullptr) {
+        throw std::runtime_error("Array cannot be null");
+    }
+
+    int numVectors = env->GetArrayLength(array2dJ);
+    this->HasExceptionInStack(env);
+
+    for (int i = 0; i < numVectors; ++i) {
+        auto vectorArray = (jbyteArray)env->GetObjectArrayElement(array2dJ, i);
```

Comment:
Please avoid c_casts and start usinhg c++ casts like static_cast or another one.